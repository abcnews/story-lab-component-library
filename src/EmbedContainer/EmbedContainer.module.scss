.embedContainer {
  margin: 0 auto 1.5rem auto;
  overflow: visible;
  width: 100%;
  clear: right;
  box-sizing: border-box;

  &.right,
  &.full {
    border: 1px solid #e6e6e6;
    padding: 1rem;
    @media (prefers-color-scheme: dark) {
      border-color: #333;
    }
  }

  @media (min-width: 34em) {
    &.right {
      float: right;
      margin-right: 1.5rem;
      width: 45%;
    }
  }

  @media (min-width: 62em) {
    &.full {
      width: auto;
      transform: translateX(-2rem);
      margin-right: -4rem;
    }
    &.right {
      width: 50%;
      margin-right: -2rem;
    }
  }
}
