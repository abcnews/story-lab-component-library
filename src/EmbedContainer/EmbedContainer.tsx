import React from "react";
import styles from "./EmbedContainer.scss";

export const EmbedContainer = ({ children, embed, className }) => {
  const cns = [styles.embedContainer];
  if (embed === "right") cns.push(styles.right);
  if (embed === "full") cns.push(styles.full);
  cns.push(className);
  return <aside className={cns.join(" ")}>{children}</aside>;
};
