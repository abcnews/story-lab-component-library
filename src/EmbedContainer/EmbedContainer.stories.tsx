import React from "react";
import { EmbedContainer } from "./EmbedContainer";

export default {
  title: "EmbedContainer"
};
const content = (
  <div>
    Here's a bunch of content inside the <code>EmbedContainer</code> component.
  </div>
);
export const Full = () => (
  <EmbedContainer embed="full">{content}</EmbedContainer>
);
export const Right = () => (
  <EmbedContainer embed="right">{content}</EmbedContainer>
);
