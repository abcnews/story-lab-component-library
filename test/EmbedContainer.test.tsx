import React from 'react';
import renderer from 'react-test-renderer'; // ES6
import { EmbedContainer } from '../src/EmbedContainer/EmbedContainer';

describe('EmbedContainer Component', () => {
  it('renders default correctly', () => {
    const tree = renderer
      .create(<EmbedContainer>Content</EmbedContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders full correctly', () => {
    const tree = renderer
      .create(<EmbedContainer embed="full">Content</EmbedContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders right correctly', () => {
    const tree = renderer
      .create(<EmbedContainer embed="right">Content</EmbedContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
