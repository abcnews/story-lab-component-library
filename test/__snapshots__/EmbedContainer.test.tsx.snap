// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedContainer Component renders default correctly 1`] = `
<aside
  className="embedContainer"
>
  Content
</aside>
`;

exports[`EmbedContainer Component renders full correctly 1`] = `
<aside
  className="embedContainer full"
>
  Content
</aside>
`;

exports[`EmbedContainer Component renders right correctly 1`] = `
<aside
  className="embedContainer right"
>
  Content
</aside>
`;
